inference-engine/include/ie_icnn_network.hpp

```diff
@@ -216,9 +215,7 @@ class INFERENCE_ENGINE_ICNNNETWORK_CLASS(ICNNNetwork) : public details::IRelease
         return NOT_IMPLEMENTED;
     }
 
-    /**
-     * @brief A virtual destructor.
-     */
-    virtual ~ICNNNetwork();
+protected:
+    ~ICNNNetwork() = default;
```

Comment:
Why did you remove virtual destructor?

Reply:
Neither virtual destruction (call destructor of derived class using pointer to base class as it is intended to be used only from `std::shared_ptr`) no RTTI (use "custom RTTI" when we check result of getFunction() method to check implementation details) So virtual destructor is not needed here.